aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -1119,6 +1139,54 @@ public <T extends S3ClientBuilder> void applyS3ServiceConfigurations(T builder)
                 .build());
   }
 
+  /**
+   * Configure a signer for an S3 client.
+   *
+   * <p>Sample usage:
+   *
+   * <pre>
+   *     S3Client.builder().applyMutation(awsProperties::applyS3SignerConfiguration)
+   * </pre>
+   */
+  public <T extends S3ClientBuilder> void applyS3SignerConfiguration(T builder) {
+    if (null != s3SignerImpl) {
+      builder.overrideConfiguration(
+          c -> c.putAdvancedOption(SdkAdvancedClientOption.SIGNER, loadS3SignerDynamically()));
+    }
+  }
+
+  @SuppressWarnings("checkstyle:NestedTryDepth")
+  private Signer loadS3SignerDynamically() {
+    // load the signer implementation dynamically
+    Object signer;
+    try {
+      signer =
+          DynMethods.builder("create")
+              .impl(s3SignerImpl, Map.class)
+              .buildStaticChecked()
+              .invoke(allProperties);
+    } catch (NoSuchMethodException e) {
```

Comment:
I think we can avoid this nesting by using this structure:

```
try {
      signer =
          DynMethods.builder("create")
              .impl(s3SignerImpl, Map.class)
              .buildStaticChecked()
              .invoke(allProperties);
    } catch (NoSuchMethodException e) {
  LOG.warn("Cannot find static method create(Map<String, String properties) for signer {}", s3SignerImpl);
}

try {
 signer = DynMethods.builder("create").impl(s3SignerImpl).buildChecked().invoke(null);
} catch (NoSuchMethodException e) {
  LOG.warn("Cannot find instance method create(Map<String, String properties) for signer {}", s3SignerImpl);
}

try {
 signer = DynMethods.builder().impl(s3SignerImpl).buildChecked().invokeInstance();
} catch (NoSuchMethodException e) {
  LOG.warn("Cannot find no-arg constructor for signer {}", s3SignerImpl);
}

if (signer == null) {
  throw ...
}
```

Reply:
I've removed the nesting and added logging as you suggested.